clients/admin-ui/src/features/datastore-connections/add-connection/ConnectorParameters.tsx

```diff
@@ -67,18 +72,24 @@ export const ConnectorParameters: React.FC<ConnectorParametersProp> = ({
           );
         }
         break;
+      case SystemType.EMAIL:
+        if (isSuccess && data) {
+          return (
+            <EmailConnectorParameters
+              data={data}
+              onConnectionCreated={handleRouteToDatastore}
+              onTestEmail={handleTestConnectionClick}
+            />
+          );
+        }
     }
-  }, [connectionOption?.type, data, isSuccess, onConnectionCreated]);
-
-  useEffect(() => {
-    mounted.current = true;
-    if (connectionOption?.type !== SystemType.MANUAL) {
-      setSkip(false);
-    }
-    return () => {
-      mounted.current = false;
-    };
-  }, [connectionOption?.type]);
```

Comment:
this `useEffect` + skip pattern was hard to follow, so I refactored it and it seems to still work well